gpt4all-backend/llmodel.cpp

```diff
@@ -90,32 +100,33 @@ const std::vector<LLModel::Implementation> &LLModel::Implementation::implementat
                 // Iterate over all libraries
                 for (const auto& f : std::filesystem::directory_iterator(fs_path)) {
                     const std::filesystem::path& p = f.path();
+
                     if (p.extension() != LIB_FILE_EXT) continue;
+                    if (!std::regex_search(p.stem().string(), re)) continue;
+
                     // Add to list if model implementation
                     try {
                         Dlhandle dl(p.string());
                         if (!Implementation::isImplementation(dl)) {
                             continue;
                         }
-                        fres.emplace_back(Implementation(std::move(dl)));
+                        s_impl_libs.emplace_back(std::make_unique<Implementation>(Implementation(std::move(dl))));
```

Comment:
I think we can avoid copy-constructing the implementation here:
```suggestion
                        s_impl_libs.emplace_back(std::make_unique<Implementation>(std::move(dl)));
```